Assets/Scripts/Structure/Door.cs

```diff
@@ -0,0 +1,113 @@
+{
+    public enum DoorType
+    {
+        Single,
+        Double
+    };
+
+    // As is the standard in the rest of the code, wallCap should face east.
+    [SerializeField]
+    private GameObject wallCapPrefab; 
+
+    [SerializeField]
+    private DoorType doorType;
+
+    /**
+     * When a single adjacent turf is updated
+     */
+    public void UpdateSingle(Direction direction, ConstructibleTile tile)
+    {
+        UpdateSingleConnection(direction, tile);
+        SetMeshAndDirection();
+    }
+
+    /**
+     * When all (or a significant number) of adjacent turfs update.
+     * Turfs are ordered by direction, i.e. North, NorthEast, East ... NorthWest
+     */
+    public void UpdateAll(ConstructibleTile[] tiles)
+    {
+        for (int i = 0; i < tiles.Length; i++) {
+            UpdateSingleConnection((Direction)i, tiles[i]);
+        }
+        SetMeshAndDirection();
+    }
+
+    protected override void OnStateUpdate(DoorState prevState)
+    {
+        float rotation = TileState.orientation == Orientation.Vertical ? 0 : 90;
+        transform.localRotation = Quaternion.Euler(0, rotation, 0);
+    }
+
+    private void OnValidate()
+    {
+        OnStateUpdate(new DoorState());
+        SetMeshAndDirection();
+    }
+
+    /**
+     * Adjusts the connections value based on the given new tile
+     */
+    private void UpdateSingleConnection(Direction direction, ConstructibleTile tile)
+    {
+        bool isConnected = tile.turf.genericType == "wall";
+
+        // Set the direction bit to isConnected (1 or 0)
+        connections &= (byte)~(1 << (int)direction);
+        connections |= (byte)((isConnected ? 1 : 0) << (int)direction);
+    }
+
+    private void SetMeshAndDirection()
+    {
+        if(wallCapPrefab == null)
+            return;
+
+        // Go through each direction and ensure the wallcap is present.
+        for(int i = 0; i < 4; i++) {
+            Direction direction = (Direction)(i * 2);
+            bool isPresent = Adjacent(direction) == 1;
+            if(isPresent && wallCaps[i] == null) {
+                wallCaps[i] = EditorAndRuntime.InstantiatePrefab(wallCapPrefab, transform);
+                var cardinal = DirectionHelper.ToCardinalVector(direction);
+                wallCaps[i].transform.localRotation = Quaternion.Euler(0, (1 - i / 2) * 180, 0);
+                wallCaps[i].transform.localPosition = new Vector3(cardinal.Item1 + cardinal.Item2, 0, 0);
+            }
+            else if(!isPresent && wallCaps[i] != null) {
+                EditorAndRuntime.Destroy(wallCaps[i]);
+            }
+        }
+    }
+
+    /**
+     * Returns 0 if no adjacency, or 1 if there is.
+     */
+    private int Adjacent(Direction direction)
+    {
+        return (connections >> (int)direction) & 0x1;
+    }
+
+    // WallCap gameobjects, North, East, South, West. Null if not present.
+    private GameObject[] wallCaps = new GameObject[4];
+    private byte connections;
```

Comment:
They're private, so I put them below anything public.
That way a reader sees the interface in full first

Reply:
OK. We'll it's definitely a style thing to agree on. I can see the reasoning behind it, but I am used to having variables at the top, so it's triggering me 😁

Reply:
Yeah I want to find whoever decided upon that style and slap them, though I suspect it was made up because of something stupid like early compilers requiring all fields come before any methods.